docs/design/run.md

```diff
@@ -0,0 +1,275 @@
+of the tree based on the parent commit. In addition to these two modes, we may 
+want to have an option to ignore any changes made in the subprocess's working 
+copy.
+
+### Modifying the Repo:
+
+Once we give the subprocess access to a fork of the repo via separate 
+[OpHeadsStore], it will be able to create new operateions in its fork.
+If the user runs `jj run -r foo` and the subprocess checks out another commit,
+it's not clear what that should do. We should probably just verify that the 
+working-copy commit's parents are unchanged after the subprocess returns. Any
+operations creeatd by the subprocess will be ignored. 
+
+### Rewriting the revisions. 
+
+We should handle public and private revisions differently. We choose to operate
+on an immutable history by default.
+
+### Public revisions:
+
+For published revisions, we will not allow `jj run?  to modify them and then 
+immediately error out, as published history should be immutable. We may want to
+support a `--force` flag for an override but it won't be available in the first
+iteration of the command. 
+
+### Private/Draft revisions:
+
+For private/draft revisions, we just amend the changes, as Jujutsu usually does. 
+We also expose the actual behavior as a command option.
+
+## Execution order/parallelism:
+
+It may be useful to execute commands in topological order. For example, 
+commands with costs proportional to incremental changes, like build systems. 
+There may also be other revelant heuristics, but topological order is an easy
+and effective way to start. 
+
+Parallel execution of commands on different commits may choose to schedule 
+commits to still reduce incremental changes in the working copy used by each
+execution slot/"thread". However, running the command on all commits 
+concurrently should be possible if desired. 
+
+Executing commands in topological order allows for more meaningful use of any 
+potential features that stop execution "at the first failure". For example, 
+when running tests on a chain of commits, it might be useful to proceed in 
+topological/chronological order, and stop on the first failure, because it 
+might imply that the remaining executions will be undesirable because they will
+also fail.
+
+## Dealing with failure:
+
+It will be useful to have multiple strategies to deal with failures on a single
+or multiple revisions. The reason for these strategies is to allow customized
+conflict handling. These strategies then can be exposed in the ui with a 
+matching command.
+
+**Continue:** If any subprocess fails, we will continue the work on child 
+revisions. Notify the user on exit about the failed revisions. 
+
+**Stop:** Signal a fatal failure and cancel any scheduled work that has not
+yet started running, but let any already started subprocess finish. Notify the
+user about the failed command and display the generated error from the 
+subprocess. 
+
+**Fatal:** Signal a fatal failure and immediately stop processing and kill any 
```

Comment:
nit: I might call this mode "Abort" instead ("Fatal" isn't a verb), but we'll revisit this when we review the patches anyway

Reply:
I left it alone for now.